core/src/test/java/org/elasticsearch/indexing/IndexActionIT.java

```diff
@@ -223,4 +229,35 @@ public void testInvalidIndexName() {
                     e.getMessage().contains("Invalid index name [..], must not be \'.\' or '..'"), equalTo(true));
         }
     }
+
+    /**
+     * This test ensures that indexing operations adhere to the {@link IndexSettings#WAIT_FOR_ACTIVE_SHARDS_SETTING}.
+     */
+    public void testChangeWaitForActiveShardsSetting() throws Exception {
+        final String indexName = "test";
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        final int numReplicas = internalCluster().numDataNodes() + randomIntBetween(0, 3);
+        final Settings originalSettings = Settings.builder()
+                                              .put(IndexMetaData.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
+                                              .put(IndexMetaData.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), numReplicas)
+                                              .build();
+        assertAcked(client().admin().indices().prepareCreate(indexName).setSettings(originalSettings).get());
+
+        // active shard count not met, should throw exception
+        client().admin().indices().prepareUpdateSettings(indexName).setSettings(
+            Settings.builder().put(WAIT_FOR_ACTIVE_SHARDS_SETTING.getKey(), Integer.toString(internalCluster().numDataNodes() + 1)).build()
```

Comment:
Done

On Fri, Jul 29, 2016 at 10:10 AM Boaz Leskes <notifications@github.com>
wrote:

> In core/src/test/java/org/elasticsearch/indexing/IndexActionIT.java
> <https://github.com/elastic/elasticsearch/pull/19454#discussion_r72797211>
> :
>
> > +    /**
> > +     * This test ensures that indexing operations adhere to the {@link IndexSettings#WAIT_FOR_ACTIVE_SHARDS_SETTING}.
> > +     */
> > +    public void testChangeWaitForActiveShardsS
>
> etting() throws Exception {
> > +        final String indexName = "test";
> > +        internalCluster().ensureAtLeastNumDataNodes(2);
> > +        final int numReplicas = internalCluster().numDataNodes() + randomIntBetween(0, 3);
> > +        final Settings originalSettings = Settings.builder()
> > +                                              .put(IndexMetaData.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
> > +                                              .put(IndexMetaData.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), numReplicas)
> > +                                              .build();
> > +        assertAcked(client().admin().indices().prepareCreate(indexName).setSettings(originalSettings).get());
> > +
> > +        // active shard count not met, should throw exception
> > +        client().admin().indices().prepareUpdateSettings(indexName).setSettings(
> > +            Settings.builder().put(WAIT_FOR_ACTIVE_SHARDS_SETTING.getKey(), Integer.toString(internalCluster().numDataNodes() + 1)).build()
>
> sometimes use an illegal negative value?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/19454/files/bcfd58f1bbac5e0dc7a65fa9365821ef0b39c052#r72797211>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABjkQVlW2m0rrYxBAqkBROOhsNqwqe2bks5qagm5gaJpZM4JNuiL>
> .
>